### ISSUE TITLE
SendRichEmail} - SendWith Attachment - Attachment related list to email is empty

### ISSUE THREAD

**Issue by edmondo1984** [alexed1/LightningFlowComponents#570]
When you use the SendRichEmail to send an email with a lightning template and attaching the file, the file is not shown in the attachment list in the email record. This is annoying, because although we know the attachment is there, we can't see

**Comment by jdpond** [comment]
Is this with saveAsActivity, saveAsTask, or both.  Also is this massEmail or singleEmail?

**Comment by edmondo1984** [comment]
saveAsActivity, and singleEmail. I think the real problem might be related to this setting
- https://trailblazer.salesforce.com/ideaView?id=08730000000sZ8DAAU
- https://help.salesforce.com/HTViewSolution?id=000199222

**Comment by jdpond** [comment]
saveAsActivity is a function provided by the Message class.  It does not surprise me that attachments are not saved.  I have not reviewed saveAsTask, but in the past, I have almost always had to create custom code for saving attachments any time Messaging.InboundEmailHandler is used, so it would surprise me if attachments are being saved without the same custom code being introduced.

Long story short - will not attempt to fix saveAsActivity - that needs to go to Salesforce.  If this is important, could look at saveAsTask in a future release.

If you're OK with that, could you close this one?  If you need saveAsTask open a new future request?  Thanks

**Comment by edmondo1984** [comment]
Thanks a lot. I am not sure I understood, can you please re-explain ? saveAsActivity cannot be fixed (Salesforce should fix that) while in saveAsTask it can work?

**Comment by jdpond** [comment]
saveAsActivity is a Message method.  saveAsTask was implemented via APEX code (part of sendBetterEmail)
[setSaveAsActivity(saveAsActivity)](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_email_outbound_base.htm#apex_Messaging_Email_setSaveAsActivity)
Optional. The default value is true, meaning the email is saved as an activity. This argument only applies if the recipient list is based on targetObjectId or targetObjectIds. If HTML email tracking is enabled for the organization, you will be able to track open rates.

**Comment by jdpond** [comment]
The problem here is the default value of the parameter is false, when the default parameter in the Messaging Class is true, and CPE automatically defaults all Boolean Checkboxes to false(NULL). The way we "fixed" this was to create a "new" component (sendBetterEmail) and documented that the default WOULD be false and that any existing usage would have to be manually edited for this. However, I'm not sure this works in all of the 7 use cases we uncovered during this process (as a matter of fact, since 'false' is presented as a string value '$GlobalConstant.False' (which would be a logical TRUE), the behavior you describe is plausible.

Since then, @ericrsmith35 came up with a solution which has been implemented for World Wide United Way and they are in the process of testing it. If this works, I can revisit the existing Boolean parameters to force parameter defaulting

Will update here after further exploration.
